tests/Collection/CollectionFunctionalTest.php

```diff
@@ -179,6 +215,40 @@ public function testFindOne()
         $this->assertSameDocument($expected, $this->collection->findOne($filter, $options));
     }
 
+    public function testFindWithinTransaction()
+    {
+        $this->skipIfTransactionsAreNotSupported();
+
+        // Collection must be created before the transaction starts
+        $options = ['writeConcern' => new WriteConcern(WriteConcern::MAJORITY)];
+        $operation = new CreateCollection($this->getDatabaseName(), $this->getCollectionName(), $options);
+        $operation->execute($this->getPrimaryServer());
+
+        try {
+            $session = $this->manager->startSession();
+            $session->startTransaction();
+
+            $this->createFixtures(3, ['session' => $session]);
+
+            $cursor = $this->collection->find(
+                ['_id' => ['$lt' => 3]],
+                ['session' => $session]
+            );
+
+            $expected = [
+                ['_id' => 1, 'x' => 11],
+                ['_id' => 2, 'x' => 22],
+            ];
+
+            $this->assertSameDocuments($expected, $cursor);
+
+            $this->commitTransaction();
+        } catch (Exception $e) {
+            $session->abortTransaction();
```

Comment:
Since the `find()` tests throw an exception, we must be careful to abort the transaction before allowing PHPUnit to process the test failure. Without doing so, we leave the transaction open on the server and a subsequent run of this test will hang when attempting to drop the collection during `setUp()`. Consider the following mongod log:

```
2018-07-13T11:40:18.069-0400 I COMMAND  [conn80] CMD: drop phplib_test.CollectionFunctionalTest.1ab43437

*** more than 60 seconds elapse ***

2018-07-13T11:41:20.741-0400 I STORAGE  [conn80] dropCollection: phplib_test.CollectionFunctionalTest.1ab43437 (88ff4330-4ee4-44b2-afcd-933f32b5d741) - renaming to drop-pending collection: phplib_test.system.drop.1531496480i1t3.CollectionFunctionalTest.1ab43437 with drop optime { ts: Timestamp(1531496480, 1), t: 3 }
2018-07-13T11:41:20.741-0400 I STORAGE  [conn80] renameCollection: renaming collection 88ff4330-4ee4-44b2-afcd-933f32b5d741 from phplib_test.CollectionFunctionalTest.1ab43437 to phplib_test.system.drop.1531496480i1t3.CollectionFunctionalTest.1ab43437
2018-07-13T11:41:20.744-0400 I COMMAND  [conn80] command phplib_test.CollectionFunctionalTest.1ab43437 command: drop { drop: "CollectionFunctionalTest.1ab43437", $db: "phplib_test", lsid: { id: UUID("d3fa5cf7-d8a2-4e9f-b764-c09c7dbaa401") }, $clusterTime: { clusterTime: Timestamp(1531496418, 4), signature: { hash: BinData(0, 0000000000000000000000000000000000000000), keyId: 0 } } } numYields:0 reslen:234 locks:{ Global: { acquireCount: { r: 2, w: 2 } }, Database: { acquireCount: { w: 1, W: 1 }, acquireWaitCount: { W: 1 }, timeAcquiringMicros: { W: 62671648 } }, oplog: { acquireCount: { w: 1 } } } protocol:op_msg 62675ms
2018-07-13T11:41:20.744-0400 I REPL     [replication-1] Completing collection drop for phplib_test.system.drop.1531496480i1t3.CollectionFunctionalTest.1ab43437 with drop optime { ts: Timestamp(1531496480, 1), t: 3 } (notification optime: { ts: Timestamp(1531496480, 1), t: 3 })
2018-07-13T11:41:20.745-0400 I STORAGE  [replication-1] Finishing collection drop for phplib_test.system.drop.1531496480i1t3.CollectionFunctionalTest.1ab43437 (88ff4330-4ee4-44b2-afcd-933f32b5d741).
```

Reply:
It looks like this is actually the result of failing to use a majority write concern during the initial drop/create of the test collection. See [PHPLIB-377](https://jira.mongodb.org/browse/PHPLIB-377).

Reply:
On second thought, it may actually be a problem with an in-progress transaction not being aborted when the `MongoDB\Driver\Session` is destroyed. This may be a libmongoc bug.

Reply:
Fourth time's the charm. After some more testing, I can attest that this is not a libmongoc bug. Rather, the exception being thrown includes a reference to the `$session` and keeps it and the transaction open beyond the scope of the test method itself.

I'll see if I can work out another approach that doesn't require explicit aborting in the meantime.

Reply:
Note: once [PHPC-1243](https://jira.mongodb.org/browse/PHPC-1243) is resolved and released, we can change this to `finally { $session->endSession(); }` (with a bump to our PHPC dependency).